core/src/main/java/io/github/tcdl/msb/collector/Collector.java

```diff
@@ -253,6 +242,22 @@ private int setResponsesRemainingForResponderId(String responderId, Integer resp
         return responsesRemainingById.get(responderId);
     }
 
+    public void waitForResponses() {
+        int newTimeoutMs = this.currentTimeoutMs - toIntExact(clock.instant().toEpochMilli() - this.startedAt);
+        LOG.debug("Waiting for responses until {}.", clock.instant().plus(newTimeoutMs, ChronoUnit.MILLIS));
+        this.responseTimeoutFuture = timeoutManager.enableResponseTimeout(newTimeoutMs, this);
+    }
+
+    void waitForAcks() {
+        if (ackTimeoutFuture == null) {
+            LOG.debug("Waiting for ack until {}.", this.waitForAcksUntil);
+            long ackTimeoutMs = waitForAcksUntil.toEpochMilli() - clock.instant().toEpochMilli();
+            ackTimeoutFuture = timeoutManager.enableAckTimeout(toIntExact(ackTimeoutMs), this);
```

Comment:
really don't got u poin. We call isAwaitingAcks() before call to this method that contains check for null. In case null we never call this method. This method is package-private for testing puropse and why do we ever need null check for testing?? Please provide more details